docs/explanation/testing.md

```diff
@@ -1,67 +1,115 @@
 
-**Tools.** Unit testing a charm can be done using:
+1. Setting up the charm, metadata, context, output mocks, and Juju state.
+2. Simulating events using `Context.run`. For example, `config_changed`, `relation_changed`, `storage_attached`, `pebble_ready`, and so on.
+3. Retrieving and asserting the output.
 
-- [`pytest`](https://pytest.org/) and/or [`unittest`](https://docs.python.org/3/library/unittest.html) and
-- [state transition testing](ops_testing), using the `ops` unit testing framework
+`Context` and `State` are instantiated before the charm. This enables you to prepare the state of config, relations, and storage before simulating an event.
 
-**Examples.**
+### Examples
 
 - [https://github.com/canonical/prometheus-k8s-operator/blob/main/tests/unit/test_charm.py](https://github.com/canonical/prometheus-k8s-operator/blob/main/tests/unit/test_charm.py)
 
+(interface-tests)=
 ## Interface testing
 
-In the context of a charm, interface tests help validate charm library behavior without individual charm code against mock Juju APIs. 
+Interface tests validate charm library behavior against mock Juju APIs, ensuring compliance with an interface specification without requiring individual charm code.
+
+Interface specifications, stored in {ref}`charm-relation-interfaces <charm-relation-interfaces>`, are contract definitions that mandate how a charm should behave when integrated with another charm over a registered interface. For information about how to create an interface, see {ref}`register-an-interface`.
+
+> See also: {ref}`write-tests-for-an-interface`.
+
+### Coverage
+
+Interface tests enable Charmhub to validate the relations of a charm and verify that your charm supports the registered interface. For example, if your charm supports an interface called "ingress", interface tests enable Charmhub to verify that your charm supports the [registered `ingress` interface](https://github.com/canonical/charm-relation-interfaces/tree/main/interfaces/ingress/v2).
+
+Interface tests also:
+- Enable alternative implementations of an interface to validate themselves against the contractual specification stored in `charm-relation-interfaces`.
+- Help verify compliance with multiple versions of an interface.
+
+An interface test has the following pattern: 
+
+1) **Given** - An initial state of the relation over the interface under test.
+2) **When** - A specific relation event fires.
+3) **Then** - The state of the databags is valid. For example, the state satisfies a [pydantic](https://docs.pydantic.dev/latest/) schema.
+
+In addition to validity of the databag state, we could check for more elaborate conditions.
 
-> See more: {ref}`interface-tests`
+### Tools
 
+- [`ops.testing`](ops_testing)
+- A pytest plugin called [`pytest-interface-tester`](https://github.com/canonical/pytest-interface-tester)
+
+### Examples
+
+A typical interface test looks like:
```

Comment:
Can we just link to an example here, rather than copy one in, to avoid cluttering this doc?